### PR TITLE
[ie/Eurosport] Support local variants

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Add local Eurosport variants. I needed to disable xff as that would make videos unavailable even when in the correct region.
Potentially in a follow-up PR the correct XFF header can be automaticly determined.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/eurosport.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 
 class EurosportIE(InfoExtractor):
-    _VALID_URL = r'https?://www\.eurosport\.com/\w+/(?:[\w-]+/[\d-]+/)?[\w-]+_(?P<id>vid\d+)'
+    _VALID_URL = r'https?://(?:(?:www\.)?eurosport\.(?:com(?:\.tr)?|de|dk|es|fr|hu|it|nl|no|ro)|eurosport.tvn24.pl)/[\w-]+/(?:[\w-]+/[\d-]+/)?[\w.-]+_(?P<id>vid\d+)'
```
Comment (grqz) [inline yt_dlp/extractor/eurosport.py:6]: Also:
`https://www.tntsports.co.uk/`
`https://espanol.eurosport.com/`
Just found out that these domain names can be retrieved from the 'CHANGE VERSION' list at the bottom of the page.
image

File: yt_dlp/extractor/tvn24.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 class TVN24IE(InfoExtractor):
     _WORKING = False
-    _VALID_URL = r'https?://(?:(?:[^/]+)\.)?tvn24(?:bis)?\.pl/(?:[^/]+/)*(?P<id>[^/]+)'
+    _VALID_URL = r'https?://(?:(?!eurosport)(?:[^/]+)\.)?tvn24(?:bis)?\.pl/(?:[^/]+/)*(?P<id>[^/]+)'
```
Comment (bashonly) [inline yt_dlp/extractor/tvn24.py:11]:
```suggestion
    _VALID_URL = r'https?://(?:(?!eurosport)[^/]+\.)?tvn24(?:bis)?\.pl/(?:[^/?#]+/)*(?P<id>[^/?#]+)'
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/eurosport.py (hunk 1 above)
Comment (seproDev) [reply yt_dlp/extractor/eurosport.py:6]: The redirect of tntsports.co.uk to eurosport.com is a geo-block. Though, even with a VPN I get 403s and 451s on the media endpoints, most likely due to using a VPN. The one video I was able to access was DRM protected.
For now, I would leave this out until someone from the UK can confirm what works and what doesn't.

I'll add `espanol.eurosport.com`. Thanks!

Comment (grqz) [reply yt_dlp/extractor/eurosport.py:6]: There are too many domain names now. I suggest creating a list of domains like in the VidflexIE pr.

Comment (bashonly) [reply yt_dlp/extractor/eurosport.py:6]: IMO a list isn't necessary, but a multiline regex pattern might be nice to break up the long line

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/eurosport.py (hunk 1 above)
Comment (grqz) [reply yt_dlp/extractor/eurosport.py]: But the current regex matches `espanol.eurosport.de`, which doesn't seem to work. Does that matter?

Comment (seproDev) [reply yt_dlp/extractor/eurosport.py]: Yep, I'm aware. I think it is fine to keep the regex a bit simpler.

Comment (bashonly) [reply yt_dlp/extractor/eurosport.py]: > I think it is fine to keep the regex a bit simpler.

agreed. it's only a problem if the user manually constructs a bogus url

**Pull request merged**